styleguide-app/Examples/Carousel.elm

```diff
@@ -0,0 +1,236 @@
+        |> Control.field "containerStyles" controlContainerStyles
+
+
+controlTabListStyles : Control ( String, List Style )
+controlTabListStyles =
+    ( "[ Css.displayFlex, Css.property \"gap\" \"20px\" ]"
+    , [ Css.displayFlex, Css.property "gap" "20px" ]
+    )
+        |> Control.value
+        |> Control.maybe False
+        |> Control.map (Maybe.withDefault ( "[]", [] ))
+
+
+controlTabStyles : Control ( String, Int -> Bool -> List Css.Style )
+controlTabStyles =
+    let
+        simplifiedCodeVersion =
+            "\\index isSelected -> [ -- styles that depend on selection status\n    ]"
+    in
+    (\_ isSelected ->
+        let
+            ( backgroundColor, textColor ) =
+                if isSelected then
+                    ( Colors.azure, Colors.white )
+
+                else
+                    ( Colors.gray92, Colors.gray20 )
+        in
+        [ Css.padding2 (Css.px 10) (Css.px 20)
+        , Css.backgroundColor backgroundColor
+        , Css.borderRadius (Css.px 8)
+        , Css.border Css.zero
+        , Css.color textColor
+        , Css.cursor Css.pointer
+        ]
+    )
+        |> Control.value
+        |> Control.maybe False
+        |> Control.map (Maybe.withDefault (\_ _ -> []))
+        |> Control.map (\v -> ( simplifiedCodeVersion, v ))
+
+
+controlContainerStyles : Control ( String, List Style )
+controlContainerStyles =
+    ( "[ Css.margin (Css.px 20) ]", [ Css.margin (Css.px 20) ] )
+        |> Control.value
+        |> Control.maybe False
+        |> Control.map (Maybe.withDefault ( "[]", [] ))
+
+
+type Msg
+    = FocusAndSelectTab { select : Int, focus : Maybe String }
+    | Focused (Result Dom.Error ())
+    | SetSettings (Control Settings)
+
+
+update : Msg -> State -> ( State, Cmd Msg )
+update msg model =
+    case msg of
+        FocusAndSelectTab { select, focus } ->
+            ( { model | selected = select }
+            , focus
+                |> Maybe.map (Dom.focus >> Task.attempt Focused)
+                |> Maybe.withDefault Cmd.none
+            )
+
+        Focused error ->
+            ( model, Cmd.none )
+
+        SetSettings settings ->
+            ( { model | settings = settings }, Cmd.none )
+
+
+moduleName : String
+moduleName =
+    "Carousel"
+
+
+version : Int
+version =
+    1
+
+
+example : Example State Msg
+example =
+    { name = moduleName
+    , version = version
+    , categories = [ Layout ]
+    , keyboardSupport =
+        [ { keys = [ KeyboardSupport.Tab ]
+          , result = "Move focus to the currently-selected Tab's tab panel"
+          }
+        , { keys = [ Arrow KeyboardSupport.Left ]
+          , result = "Select the tab to the left of the currently-selected Tab"
+          }
+        , { keys = [ Arrow KeyboardSupport.Right ]
+          , result = "Select the tab to the right of the currently-selected Tab"
+          }
+        ]
+    , state = init
+    , update = update
+    , subscriptions = \_ -> Sub.none
+    , preview =
```

Comment:
If it's possible to add a little widget preview, this is where it would slot in

<img width="208" alt="" src="https://user-images.githubusercontent.com/8811312/173964379-cd37ea8c-fce8-4796-a387-d84a713b9a3d.png">

Reply:
Will do! 🎉